### PR TITLE
feat: dynamic property for ImageURLOptions

### PR DESCRIPTION
**Pull request by Tenpi**
**Please describe the changes this PR makes and why it should be merged:**
I added a property named `dynamic` to `ImageURLOptions` which will return a `gif` url if the image is animated, and a url in the type of format you specified otherwise (`webp` if nothing is specified). `webp` images don't display on my phone, so personally I would prefer to have a `png` link instead. 

Example:
```ts
//Returns gif url if the icon is animated, png otherwise. 
message.guild.iconURL({format: "png", dynamic: true}) 
```
Again maybe I'm wrong, but `AvatarOptions` in the typings seems to refer to the same thing as `ImageURLOptions`, so I renamed it since it's the name used in the docs. 

**Status**
- [X] Code changes have been tested against the Discord API, or there are no code changes
- [X] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [X] This PR changes the library's interface (methods or parameters added)
  - [X] This PR includes breaking changes (methods removed or renamed, parameters moved or removed) (Minor)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by kyranet** [comment]
I think it's much less opinionated and much easier to do `{ format: 'webp' }` instead of `{ dynamic: false }`. The name `dynamic` doesn't describe this behaviour correctly.

**Comment by Monbrey** [comment]
If I'm understanding this correctly - you want a way to retrieve an animated avatar while also choosing the fallback format, instead of the default always being `webp`?

**Comment by DevYukine** [comment]
This change seems unreasonable to me, its not only making it unecessary unintuitiv but also you didn't change anything at all, the default is still `webp`. i can see why you would want to use png in your case but then just store an object holding your default values and re-use it inside your own code 🤔.

**Comment by Tenpi** [comment]
> I think it's much less opinionated and much easier to do `{ format: 'webp' }` instead of `{ dynamic: false }`. The name `dynamic` doesn't describe this behaviour correctly.

Yes you're right, should be false by default. I want to do something like 
```ts
message.guild.iconURL({format: "png", dynamic: true}) 
```
Which will return a gif whenever the iconURL is animated, and a png if it is not. Currently you can't customize the fallback format, it will always be `webp`. (I think that it should be `png` by default anyway, but that is my opinion).

File: typings/index.d.ts [hunk 1]
```diff
@@ -1374,10 +1374,10 @@ declare module 'discord.js' {
 		public readonly presence: Presence;
 		public readonly tag: string;
 		public username: string;
-		public avatarURL(options?: AvatarOptions): string | null;
+		public avatarURL(options?: ImageURLOptions & {dynamic?: boolean}): string | null;
```
Comment (iCrawl) [inline typings/index.d.ts]:
```suggestion
		public avatarURL(options?: ImageURLOptions & { dynamic?: boolean }): string | null;
```

File: typings/index.d.ts [hunk 2]
```diff
@@ -1374,10 +1374,10 @@ declare module 'discord.js' {
...
 		public createDM(): Promise<DMChannel>;
 		public deleteDM(): Promise<DMChannel>;
-		public displayAvatarURL(options?: AvatarOptions): string;
+		public displayAvatarURL(options?: ImageURLOptions & {dynamic?: boolean}): string;
```
Comment (iCrawl) [inline typings/index.d.ts]:
```suggestion
		public displayAvatarURL(options?: ImageURLOptions & { dynamic?: boolean }): string;
```

**Pull request merged**